packages/core/src/checkout/checkout-service.ts

```diff
@@ -1399,6 +1405,30 @@ export default class CheckoutService {
         return this._dispatch(action, { queueId: 'extensions' });
     }
 
+    /**
+     * Registers an extension's command handlers.
+     *
+     * @alpha
+     * @param extensionId - The ID of an extension sending the commands.
+     * @param handlers - A set of handlers for the extension commands.
+     */
+    addExtensionCommandHandlers(extensionId: string, handlers: ExtensionCommandHandlers): void {
```

Comment:
I think it's better to return a function that could unsubscribe the handler, similar to the existing `subscribe` method. e.g.:

```ts
listenExtensionCommand(extensionId: string, commandType: ExtensionCommand, commandHandler, ExtensionCommandHandler): () => void
```

This makes it easier to unsubscribe a handler, as calling the return value can specifically remove such handler. Otherwise, clients need to hold a reference to the handler and therefore can't use inline anonymous function as the handler.

Reply:
I have updated it as:
```
listenExtensionCommand(
        extensionId: string,
        command: ExtensionCommand,
        handler: ExtensionCommandHandler,
    ): () => void
```
To entirely remove the command listening, I also added a method in the `checkoutService` to stop listening for an extension.
```
stopListenExtensionCommand(extensionId: string): void
```